docs/GettingStartedDocs/Contributors/SignUsingEngines.md

```diff
@@ -0,0 +1,113 @@
+# Signing a Build With OpenSSL Engines
```

Comment:
Thanks for writing this up, Bruce, it's pretty educational.

A lot of it is unrelated to OE SDK though (e.g. the discussion about the default set of engines could just be linked to OpenSSL's README.ENGINE, which looks like where you go it from anyway); it's unclear to me if we're trying to communicate broader support for OpenSSL Engines than we've actually decided to expose through the oesign tool. 

What are we trying to communicate here beyond the usage command?

Reply:
OpenSSL documentation for engines is epically bad.   It uses a lot of vague descriptions with words like "may" or "could" and  leaves one very little practical info to work with.  There are a couple of OK descriptions of engines if you do a sufficiently deep web search and read through all of the options, but basically there is very little good doc on the subject.

The doc was intended to communicate:
1. What is an engine, in specific terms, and what does it do.
2. If someone picking up the SDK wanted to use and engine, what would be realistic to expect it to do.
3. If someone picking up the SDK liked the idea of using an engine for their environment (which I believe is to be encouraged), how would they implement one, though clearly one could only sort of orient peoplem not provide complete directions.

To the end of (3), we also provide an example in the form of oesign-test-engine which doesn't do much but is a complete, functioning engine.

Again, there is no good doc on the subject on the net.